rero_ils/modules/holdings/api.py

```diff
@@ -241,6 +243,90 @@ def get_holding_loan_conditions(self):
+        return self.get('patterns')
+
+    @property
+    def next_issue_display_text(self):
+        """Display the text of the next predicted issue."""
+        if not self.patterns:
+            return
+        return self._get_next_issue_display_text(self.patterns)
+
+    @classmethod
+    def _get_next_issue_display_text(cls, patterns):
+        """Display the text for the next predicted issue.
+
+        :param patterns: List of a valid holdings patterns.
+        :returns: A display text of the next predicted issue.
+        """
+        texts = []
+        for pattern in patterns:
+            text = []
+            for key, levels in pattern.items():
+                if key == 'levels':
+                    for level in levels:
+                        show_level = level.get('show', True)
+                        if show_level:
+                            text_value = level.get('next_value', level.get(
+                                'starting_value', 1))
+                            mapping = level.get('mapping_values')
+                            if mapping:
+                                text_value = mapping[text_value - 1]
+                            text.append(
+                                ' '.join(
+                                    (level.get('name', ''), str(text_value))
+                                ).lstrip())
+            texts.append(' '.join(text))
+        return ' '.join(texts)
+
+    def increment_next_prediction(self):
+        """Increment next prediction."""
+        if not self.patterns:
+            return
+        self['patterns'] = self._increment_next_prediction(self.patterns)
+
+    @classmethod
+    def _increment_next_prediction(cls, patterns):
+        """Increment the next predicted issue.
+
+        :param patterns: List of a valid holdings patterns.
+        :returns: The updated patterns with the next issue.
+        """
+        for pattern in patterns:
+            for key, levels in pattern.items():
+                if key == 'levels':
+                    for level in reversed(levels):
+                        max_value = level.get('completion_value')
+                        if level.get('mapping_values'):
+                            max_value = len(level.get('mapping_values'))
+                        next_value = level.get('next_value', level.get(
+                            'starting_value', 1))
+                        if max_value == next_value:
+                            level['next_value'] = level.get(
+                                'starting_value', 1)
+                        else:
+                            level['next_value'] = next_value + 1
+                            break
+        return patterns
+
+    def prediction_issues_preview(self, predictions=1):
+        """Display prview of next predictions.
+
+        :param predictions: Number of the next issued to predict.
+        :returns: A display text of the next predicted issues.
```

Comment:
Adapt doc string to specify that you return an array, not a simple string (as I understand when i read the docstring)